File: InsertionSort.js
let arranjo = [5,7,8,2,4,6,3]

const IsertionSort= function (arr) {
    if( arr.length == 1) return 
    let chave= arr[1]
    for(let i =1;i<arr.length;i++){
        let j = i-1
        while(arr[j]>chave && j>=0){
            arr[j+1]=arr[j]
            arr[j]= chave;
            j--;
        }
    chave= arr[i+1]
    }
}

console.log('============= DESORDENADO =============');
console.log(arranjo);
IsertionSort(arranjo)
console.log('================ ORDENADO ==========');
console.log(arranjo);

File: Algoritmos de ordenação e busca/Insertion Sort/insertionSort.py
arranjo=[5,7,8,2,4,6,3]

def InsertionSort(arranjo):
    if(len(arranjo) == 1):return
    for i in range(1,len(arranjo)):
        j=i-1
        chave = arranjo[i]
        while(arranjo[j]> chave and j>=0):
            aux = arranjo[j]
            arranjo[j] = arranjo[j+1]
            arranjo[j+1] = aux
            j-=1
        
print("Desordenado")
print(arranjo)
InsertionSort(arranjo)
print("Ordenado")
print(arranjo)
